server/coordinator/scheduler/scheduler_manager.go

```diff
@@ -182,3 +198,11 @@ func (m *ManagerImpl) Scheduler(ctx context.Context, clusterSnapshot metadata.Sn
 	}
 	return results
 }
+
+func (m *ManagerImpl) UpdateDeployMode(_ context.Context, deployMode bool) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	m.deployMode = deployMode
+	log.Info("scheduler manager update deploy mode", zap.Bool("deployMode", deployMode))
```

Comment:
Maybe we should log outside the lock here:
```suggestion
	m.lock.Lock()
	m.deployMode = deployMode
	m.lock.Unlock()


	log.Info("scheduler manager update deploy mode", zap.Bool("deployMode", deployMode))
```